File: hw4/QueueDriver.cpp
// Author: Logan Tillman

#include <cstdio>
#include "Queue.h"

int main() {
	Queue *queue = new ArrayQueue;

	queue->add(3);
	queue->add(10);
	queue->add(5);

	printf("%d\n%d", queue->remove(), queue->remove());

	queue->add(6);
	queue->add(9);
	queue->add(12);

	while (!queue->isEmpty()) {
		printf("%d\n", queue->remove());
	}
	return 0;
}

File: hw5/QueueDriver.java
// Author: Logan Tillman

import java.util.Scanner;

public class QueueDriver {
    QueueDriver() {
        Scanner console = new Scanner(System.in);
        Queue<String> myQueue = new Queue<String>(console.nextInt());

        while(console.hasNextLine()) {
            Scanner lineTokenizer = new Scanner(console.nextLine());
            String type = "";

            if (lineTokenizer.hasNext()) {
                type = lineTokenizer.next();
            }

            if (type.equals("enqueue")) {
                String item = lineTokenizer.next();
                try {
                    myQueue.enqueue(item);
                } catch(QueueOverflowException err) {

                }
            }
            else if (type.equals("dequeue")) {
                try {
                    String popped = myQueue.dequeue();
                    System.out.printf("dequeue: %s%n", popped);
                } catch(QueueEmptyException err) {

                }
            }
            else if (type.equals("print")) {
                System.out.println("queue contents");
                myQueue.printQueue();
            }

            lineTokenizer.close();
        }

        console.close();
    }

    public static void main(String[] args) {
        new QueueDriver();
    }
}
